src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationName/DeclarationNameCompletionProvider.cs

```diff
@@ -24,13 +25,14 @@ namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers
     [Shared]
     internal partial class DeclarationNameCompletionProvider : LSPCompletionProvider
     {
-        private ImmutableArray<DeclarationNameRecommender> Recommenders { get; }
+        private ImmutableArray<IDeclarationNameRecommender> Recommenders { get; }
 
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public DeclarationNameCompletionProvider()
+        public DeclarationNameCompletionProvider(PythiaDeclarationNameRecommender pythiaRecommender)
```

Comment:
don't love references to pythia from this layer outwards.  can we isntead just have a MEF importing system and have a pythia impl of that interface that just gets agnostically pulled in here?

Reply:
Sure. I did it this way since I felt what you suggested might be an overkill if it's only for Pythia